### DISCUSSION TITLE
Typescript and multiple field arrays?

### DISCUSSION THREAD

**Discussion by bwong365** [react-hook-form/react-hook-form#7583] [Q&A]
Hi, hoping to get some Typescript help with RHF&mdash;I'm getting a couple errors trying to use multiple FieldArrays.
The first error is here:
```TS
const {
  fields: personalAddressFields,
  append: addAddress,
  remove: removeAddress,
} = useFieldArray({
  control,
  name: 'personalAddresses',
});

const {
  fields: emailFields,
  append: addEmail,
  remove: removeEmail,
} = useFieldArray({
  control,
  name: 'emails', // Type '"emails"' does not satisfy the constraint '"personalAddresses"'.ts(2344)
});
```
 
Second error seems to be related to the first - I have a button that appends an email (string), but typescript seems to be inferring from the first useFieldArray only, and thinks I should be appending an address

I've tried adding the generic arguments, something like:
```TS
useFieldArray<AddressObj, "personalAddresses", "id">()

useFieldArray<string, "emails", "id">()
```
but no luck so far.

Thanks for any help everyone!

**Comment by Moshyfawn** [comment]
Can you modify [this CSB](https://codesandbox.io/s/multiple-field-arrays-example-8v4fn) to closer match what your use-case is?

**Comment by bjrice88** [comment]
How are you instantiating the useForm hook to get the control?

**Comment by bwong365** [comment]
Never mind, I see the issue is that rhf doesn't support flat arrays (which the email list was). Thanks all.
